### ISSUE TITLE
Guard against rebox

### ISSUE THREAD

**Issue by RuixiZhang42** [latex3/latex2e#517]
## Brief outline of the bug

TeX the program has a *rebox* subroutine, which is used when constructing (1)&nbsp;fraction&rsquo;s numerator or denominator, and (2)&nbsp;big operator&rsquo;s limits (and the operator itself). This subroutine does not retain the height and depth of the content box, so after reboxing these data are lost and can lead to bad interactions with some plain TeX macros (e.g., `\smash`, `\phantom`, etc.). A minimal plain TeX example is:
```tex
\def\drawbaseline{\rlap{\vrule width 50pt height 0.1pt depth 0pt }}
\noindent
This is okay: $${1\over\drawbaseline2}+{1\over\smash{2^2}}$$
This is not: $${100\over\drawbaseline2}+{100\over\smash{2^2}}$$
\bye
```
I&nbsp;reported this issue to Karl Berry and it was forwarded to Don Knuth for the 2021 tuneup. Unfortunately, according to Karl, Don Knuth declined to offer any fixes to either the program or the plain TeX format, with no explanations. Karl said that the same problems could &ldquo;leak&rdquo; into LaTeX and advised me to report them here.

## Minimal example showing the bug

```latex
%\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{article}
\newcommand*\drawbaseline{\rlap{\vrule width 50pt height 0.1pt depth 0pt }}
\begin{document}
\noindent
This is okay:
\[
\frac{1}{\drawbaseline2}+\frac{1}{\smash{2^2}}
\]
This is not:
\[
\frac{100}{\drawbaseline2}+\frac{100}{\smash{2^2}}
\]
\end{document}
```

## Log file (required) and possibly PDF file

I&nbsp;can&rsquo;t get LaTeX to run with `\RequiarePackage{latexbug}` because it keeps saying `l3backend is managed by third party`, sorry&hellip; Not a good time for update either because other deadlines.

**Comment by josephwright** [comment]
> I can’t get LaTeX to run with `\RequiarePackage{latexbug}` because it keeps saying `l3backend is managed by third party`, sorry

Not related to the report, but for me with an up-to-date `latexbug` there is no error with your demo.

**Comment by muzimuzhi** [comment]
Just to make things linked, it's resolved by #914.
